tests/components/dyson/test_sensor.py

```diff
@@ -1,355 +1,191 @@
+            "INV" if combi else values["carbon_filter_state"]
+        )
+        device.environmental_state.humidity = values["humidity"]
+        device.environmental_state.temperature = values["temperature_kelvin"]
+    else:  # DysonPureCoolLink
+        device.state.filter_life = values["filter_life"]
+        device.environmental_state.dust = values["dust"]
+        device.environmental_state.humidity = values["humidity"]
+        device.environmental_state.temperature = values["temperature_kelvin"]
+        device.environmental_state.volatil_organic_compounds = values["air_quality"]
+
+
+@callback
+def get_device(spec: Type[DysonPureCoolLink], combi=False) -> DysonPureCoolLink:
+    """Return a device of the given type."""
+    device = get_basic_device(spec)
+    _async_assign_values(device, combi=combi)
     return device
 
 
-def _get_with_standby_monitoring():
-    """Return a valid device with state but with standby monitoring disable."""
-    device = mock.Mock()
-    load_mock_device(device)
-    device.name = "Device_name"
-    device.environmental_state.humidity = 0
+@callback
+def _async_get_entity_id(sensor_type: str) -> str:
+    """Get the expected entity id from the type of the sensor."""
+    sensor_name = SENSOR_NAMES[sensor_type]
+    entity_id_suffix = sensor_name.lower().replace(" ", "_")
+    return f"{ENTITY_ID_PREFIX}_{entity_id_suffix}"
+
+
+@pytest.mark.parametrize(
+    "device,sensors",
+    [
+        (
+            DysonPureCoolLink,
+            ["filter_life", "dust", "humidity", "temperature", "air_quality"],
+        ),
+        (
+            DysonPureCool,
+            ["hepa_filter_state", "carbon_filter_state", "humidity", "temperature"],
+        ),
+        (
+            [DysonPureCool, True],
+            ["combi_filter_state", "humidity", "temperature"],
+        ),
+    ],
+    indirect=["device"],
+)
+async def test_sensors(
+    hass: HomeAssistant, device: DysonPureCoolLink, sensors: List[str]
+) -> None:
+    """Test the sensors."""
+    # Temperature is given by the device in kelvin
+    # Make sure no other sensors are set up
+    assert len(hass.states.async_all()) == len(sensors)
+
+    er = await entity_registry.async_get_registry(hass)
+    for sensor in sensors:
+        entity_id = _async_get_entity_id(sensor)
+
+        # Test unique id
+        assert er.async_get(entity_id).unique_id == f"{SERIAL}-{sensor}"
+
+        # Test state
+        state = hass.states.get(entity_id)
+        assert state.state == str(MOCKED_VALUES[sensor])
+        assert state.name == f"{NAME} {SENSOR_NAMES[sensor]}"
+
+        # Test attributes
+        attributes = state.attributes
+        for attr, map in {
+            ATTR_UNIT_OF_MEASUREMENT: {"temperature": TEMP_CELSIUS, **SENSOR_UNITS},
+            ATTR_ICON: SENSOR_ICONS,
+            ATTR_DEVICE_CLASS: SENSOR_DEVICE_CLASSES,
+        }.items():
+            expected = map.get(sensor)
+            if expected is None:
```

Comment:
Can we add the expected attributes as a parameter to the test so we don't need this logic?

Reply:
This is not easy to do since different sensors missing different attributes. Maybe add a new dict mapping sensor types to the attributes each of them should have?

Reply:
Sounds good!